src/psyclone/domain/lfric/lfric_extract_driver_creator.py

```diff
@@ -924,13 +1009,27 @@ def get_driver_as_string(self, nodes, read_write_info, prefix, postfix,
         sorted_modules = mod_manager.sort_modules(module_dependencies)
 
         # Inline all required modules into the driver source file so that
-        # it is stand-alone:
+        # it is stand-alone. Additionally, we need to remove all private
+        # declarations (since then they default to be public, which is
+        # required in order to potentially initialise an otherwise protected
+        # module variable from the data file). And similarly remove all
+        # 'protected' attributes.
+        # TODO #2142: if the LFRic build system pre-processes all files,
+        # we can modify the fparser tree or PSyIR information to do this
+        # without risking an issue if the program contains a symbol named
+        # 'protected' or 'private'.
         out = []
+        # An optional comma and spaces, followed by either protected
+        # or private as word:
+        remove_regex = re.compile(r"(, *)?(\b(protected|private)\b)")
```

Comment:
I've opened #2536 for updating documentation etc when FAB becomes the default LFRic build system. With FAB I am using an fparser script to make sure I remove the right symbols (before PSyclone is run, so extraction and driver will work). We can discuss if we want to include support for this in PSyclone itself when I tackle #2536.

Reply:
Thanks. Please could you also check what Stylist does and add this to the comment (as a justification).

Reply:
The stylist happily accepts private or protected as variable name (as does gfortran).
BUT: I have decided to just remove this whole code. I have opened #2580 to discuss if we actually have any need to do this anymore, since we can now switch to FAB (and a separate phase which can just remove the private/protected attributes everywhere). I believe we could just close #2580, but if you see anything useful in this feature, I do discuss a proper approach there.